### PR TITLE
feat(ts-bindings): upgrade stellar-sdk to 12.2.0

### PR DESCRIPTION
**Pull request by chadoh**
### What

Upgrade the version of stellar-sdk-js used by TS Bindings-generated libraries

### Why

This is the latest version

### Known limitations

[TODO or N/A]

### CODE REVIEW

**Review by fnando — APPROVED** [review]

File: cmd/crates/soroban-spec-typescript/src/project_template/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   },
   "dependencies": {
     "buffer": "6.0.3",
-    "@stellar/stellar-sdk": "12.1.0"
+    "@stellar/stellar-sdk": "12.2.0"
```
Comment (Shaptic) [inline cmd/crates/soroban-spec-typescript/src/project_template/package.json:12]: I just dropped [`v12.3.0`](https://github.com/stellar/js-stellar-sdk/releases/tag/v12.3.0), sorry :laughing:

Comment (chadoh) [reply cmd/crates/soroban-spec-typescript/src/project_template/package.json]: Oh great! Updated for 12.3.0

**Review by Shaptic — APPROVED** [review]
There was a `make` command that modified a bunch of generated files in the past; I see a smaller diff than I used do, so I'm just making sure that's not a thing anymore. Otherwise LGTM! :+1:

**Comment by leighmcculloch** [comment]
@chadoh Is this good to merge?

**Comment by chadoh** [comment]
Re-ran `make build-snapshot` to make sure and one other file ended up in the checked-in build-artifact folder. It's ready now!

**Review by fnando — APPROVED** [review]

**Comment by fnando** [comment]
@chadoh this needs to be rebased, due to #1155 being merged.

**Pull request merged**